core/sql/optimizer/NAClusterInfo.cpp

```diff
@@ -350,14 +244,8 @@ NAClusterInfo::NAClusterInfo(CollHeap * heap)
         if ((nodeInfo[i].type & MS_Mon_ZoneType_Aggregation) != 0 ||
```

Comment:
Yes, I would say we should look at the node type. Here is what I found in file core/sqf/export/include/seabed/ms.h:

    MS_Mon_ZoneType_Undefined   = 0x0000,   // No zone type defined
    MS_Mon_ZoneType_Edge        = 0x0001,   // Zone of service only nodes
    MS_Mon_ZoneType_Aggregation = 0x0002,   // Zone of compute only nodes
    MS_Mon_ZoneType_Storage     = 0x0004,   // Zone of storage only nodes
    MS_Mon_ZoneType_Excluded    = 0x0010,   // Excluded cores

Maybe we should only look at nodes of type "MS_Mon_ZoneType_Aggregation", just like NAClusterInfo does it? If you agree I can make that change.